src/Idle.DataAccess/IRequirement.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Idle.DataAccess.Fields;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Idle.DataAccess
+{
+    /// <summary>
+    /// Interface that defines requirements for accessing something
+    /// </summary>
+    public interface IRequirement
```

Comment:
hmm. whenever we would implement this interface we have to implement the members as well.
But what if something has a language requirement and not a tool requirement?
Will this ever be a case? We have to think about this.

Reply:
When it doesnt have a requirement then I thought we could just provide an empty set.

Reply:
Yes this makes sense.